spec/index.bs

```diff
@@ -2883,57 +2964,79 @@ The {{GPUMapMode}} flags determine how a {{GPUBuffer}} is mapped when calling
 
             **Returns:** {{Promise}}&lt;{{undefined}}&gt;
 
-            Issue: Figure out exactly how [=invalid=] buffers are handled here.
+            <ol data-timeline=content>
+                1. Let |contentTimeline| be the current [=Content timeline=].
+                1. Let |p| be a new {{Promise}}.
+                1. Set |this|.{{GPUBuffer/[[mapping]]}} to |p|.
+                1. Set |this|.{{GPUBuffer/[[state]]}} to [=buffer state/mapping pending=].
+                1. Set |this|.{{GPUBuffer/[[map_mode]]}} to |mode|.
+                1. Issue the |device timeline steps to validate| on the [=Device timeline=] of
+                    |this|.{{GPUObjectBase/[[device]]}}.
+                1. Return |p|.
+            </ol>
 
-            1. If |size| is missing:
-                1. Let |rangeSize| be max(0, |this|.{{GPUBuffer/size}} - |offset|).
+            The |device timeline steps to validate| are:
```

Comment:
There's something that feels a little strange about how these algorithms are broken up, but it's hard for me to put my finger on it. It makes it feel more busy than I think is strictly necessary.

Maybe simply moving the "title" of the steps inside the timeline border would help?